### DISCUSSION TITLE
Allow Cascades while deleting or updating rows

### DISCUSSION THREAD

**Discussion by NiketanG** [supabase/supabase#12330] [Feature Requests]
# Feature request

## Is your feature request related to a problem? Please describe.
Right now, if we try to delete a row which is referenced in another table by foreign key, it results in an error. An optional cascade option should be available in case referencing rows are also to be deleted.

## Describe the solution you'd like
An option, like a toggle should be given in the Supabase Dashboard when creating and editing tables, that allows to enable/disable cascades for foreign keys.

## Describe alternatives you've considered

Deleting for referenced columns manually, but this required more calls to the api/database.

**Comment by dreinon** [comment]
Hey! What's the status about this issue? Is there any work going on about this?

**Comment by madeleineostoja** [comment]
Also waiting for this, but just to clarify the original issue, there is an alternative by creating or updating tables with SQL. Postgres already supports delete cascades on foreign keys, Supabase just needs to add an option for it in the table editor UI

**Comment by riderx** [comment]
If you need to update all your past key I found this guy who list all them and create all the update request for you, copy and run all to update your keys :
https://stackoverflow.com/a/871124

**Comment by jdgamble555** [comment]
It looks they added CASCADE DELETE to the UI, but they did NOT add CASCADE UPDATE.

J

**Reply by MildTomato** [reply]
cc @joshenlim

**Comment by joshenlim** [comment]
Hey everyone! Sorry we forgot to update this discussion here but for those who've been waiting for cascade updates - wait no more!
We've added cascade updates support in the table editor right beside (or above to be exact) where cascade deletes are.

![image](https://github.com/supabase/supabase/assets/19742402/23b14023-0ffc-4ee0-b53a-b2f1b9c62874)

Hopefully this helps unblock you guys from building! I'll go ahead to close this discussion then but please do feel free to let us know if you'd like to see anything else 😄 we're always listening 👂🏻
